core/src/main/java/forklift/consumer/Consumer.java

```diff
@@ -483,10 +405,154 @@ else if (producer.topic().length() > 0)
         return closeMe;
     }
 
+    private Object constructMessageHandlerInstance(ForkliftMessage forkliftMessage, List<Closeable> closeables) throws IllegalAccessException, InvocationTargetException, InstantiationException, IOException {
+        Object instance = null;
+        if (this.constructor != null) {
+            Object[] constructorParameters = buildConstructorParameters(forkliftMessage, closeables);
+            instance = this.constructor.newInstance(constructorParameters);
+        } else {
+            instance = msgHandler.newInstance();
+        }
+        return instance;
+    }
+
+    private Object[] buildConstructorParameters(ForkliftMessage forkliftMessage, List<Closeable> closeables) throws IOException {
+        Object[] parameters = new Object[constructorAnnotations.length];
+        int index = 0;
+        for (Annotation[] parameterAnnotations : constructorAnnotations) {
+            //there must be at least one annotation that indicates what we should inject
```

Comment:
Except not, right? If there's no annotation you still handle it fine by treating it like it has `Inject`